### ISSUE TITLE
perobs version update

### ISSUE THREAD

**Issue by davidwul** [scrapper/postrunner#18]
I'm using postrunner 0.10.1 and want to update to the latest. But when I install it, it install the latest version of perobs and there is an error because of my data in v0.10.1.
What is the solution of this?

Thanks for your help

**Comment by scrapper** [comment]
Updates from databases generated by older perobs version should happen automatically the first time you open the database with a newer version. What error message do you get?

**Comment by davidwul** [comment]
Here is my error.
`ERROR: The upgrade of this version of the PEROBS database is not supported by this version of PEROBS. Please try an earlier version of PEROBS to upgrade the database before using this version.
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Log.rb:107:in `fatal'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/FlatFileDB.rb:245:in `check_version_and_upgrade'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/FlatFileDB.rb:65:in `initialize'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Store.rb:149:in `new'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Store.rb:149:in `initialize'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:59:in `new'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:59:in `main'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:24:in `<module:PostRunner>'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:22:in `<top (required)>'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/bin/postrunner:4:in `<top (required)>'
/usr/local/bin/postrunner:23:in `load'
/usr/local/bin/postrunner:23:in `<main>'

*******************************************************************************
You have triggered a bug in PostRunner 0.11.0!
`

and with the debug enable:

`Traceback (most recent call last):
	20: from /usr/local/bin/postrunner:23:in `<main>'
	19: from /usr/local/bin/postrunner:23:in `load'
	18: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/bin/postrunner:4:in `<top (required)>'
	17: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	16: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	15: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:20:in `<top (required)>'
	14: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	13: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	12: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:22:in `<top (required)>'
	11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 9: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FitFileStore.rb:19:in `<top (required)>'
	 8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FFS_Device.rb:14:in `<top (required)>'
	 5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 3: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FFS_Activity.rb:16:in `<top (required)>'
	 2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	 1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require': cannot load such file -- postrunner/ActivitySummary (LoadError)
`
with postrunner 0.10.1 (and perobs 4.00) it's running ok.
Can I provide anything else usefull?

David

**Comment by davidwul** [comment]
Here is my error.
`ERROR: The upgrade of this version of the PEROBS database is not supported by this version of PEROBS. Please try an earlier version of PEROBS to upgrade the database before using this version.
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Log.rb:107:in `fatal'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/FlatFileDB.rb:245:in `check_version_and_upgrade'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/FlatFileDB.rb:65:in `initialize'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Store.rb:149:in `new'
/var/lib/gems/2.5.0/gems/perobs-4.1.0/lib/perobs/Store.rb:149:in `initialize'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:59:in `new'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:59:in `main'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:24:in `<module:PostRunner>'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:22:in `<top (required)>'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/var/lib/gems/2.5.0/gems/postrunner-0.11.0/bin/postrunner:4:in `<top (required)>'
/usr/local/bin/postrunner:23:in `load'
/usr/local/bin/postrunner:23:in `<main>'

*******************************************************************************
You have triggered a bug in PostRunner 0.11.0!
`

and with the debug enable:

`Traceback (most recent call last):
	20: from /usr/local/bin/postrunner:23:in `<main>'
	19: from /usr/local/bin/postrunner:23:in `load'
	18: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/bin/postrunner:4:in `<top (required)>'
	17: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	16: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	15: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner.rb:20:in `<top (required)>'
	14: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	13: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	12: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/Main.rb:22:in `<top (required)>'
	11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 9: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FitFileStore.rb:19:in `<top (required)>'
	 8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FFS_Device.rb:14:in `<top (required)>'
	 5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 3: from /var/lib/gems/2.5.0/gems/postrunner-0.11.0/lib/postrunner/FFS_Activity.rb:16:in `<top (required)>'
	 2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	 1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require': cannot load such file -- postrunner/ActivitySummary (LoadError)
`
with postrunner 0.10.1 (and perobs 4.00) it's running ok.
Can I provide anything else usefull?

David

**Comment by taziden** [comment]
I'm having the same issue.
So far, I've been using postrunner 0.10.1.
Both upgrading to 0.11.0 or 0.12.0 lead to the same issue.
Edit: version file (.postrunner/database/version) contains '2' which triggers the corresponding error in perobs.

**Comment by scrapper** [comment]
/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in require': cannot load such file -- postrunner/ActivitySummary (LoadError)

That's a strange error. It can't seem to find one of the postrunner source files. Since it can file FFS_Activity.rb which is in the same directory, I would expect a permission problem. Check that ActivitySummary.rb is present and readable for you.

**Comment by taziden** [comment]
I don't think this issue should be closed.
I only resolved it by installing manually perobs/postrunner at some precise commit back in time.

I suggest you check the upgrade path between versions, it doesn't work as you described.
Using a recent version of perobs/postrunner, it's impossible to upgrade a db in version "2" (except by doing what I did, checking out old commits and manually edit stuff). It's not automagic at all.
